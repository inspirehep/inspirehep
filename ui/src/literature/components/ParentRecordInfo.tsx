import React from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import { Link } from 'react-router-dom';
import { getRecordIdFromRef } from '../../common/utils';
import { LITERATURE } from '../../common/routes';

function ParentRecordInfo({ parentRecord, publicationInfo }) {
  const pageStart = publicationInfo
    ? publicationInfo.get(0, {}).get('page_start')
    : null;
  const pageEnd = publicationInfo
    ? publicationInfo.get(0, {}).get('page_end')
    : null;
  return (
    <div>
      <span>Part of </span>
      <Link
        to={`${LITERATURE}/${getRecordIdFromRef(
          parentRecord.getIn(['record', '$ref'])
        )}`}
      >
        {parentRecord.get('title')}
        {parentRecord.has('subtitle') && (
          <span> : {parentRecord.get('subtitle')}</span>
        )}
      </Link>
      {pageStart && pageEnd && (
        <span>
          , {pageStart}-{pageEnd}
        </span>
      )}
    </div>
  );
}

ParentRecordInfo.propTypes = {
  parentRecord: PropTypes.instanceOf(Map).isRequired,
};

export default ParentRecordInfo;
