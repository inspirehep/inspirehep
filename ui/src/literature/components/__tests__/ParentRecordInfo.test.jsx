import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ParentRecordInfo from '../ParentRecordInfo';

describe('ParentRecordInfo', () => {
  it('renders with parent record', () => {
    const parentRecord = fromJS([
      {
        title: 'A title of book',
        record: { $ref: 'http://localhost:5000/api/literature/1234' },
      },
    ]);
    const wrapper = shallow(<ParentRecordInfo parentRecord={parentRecord} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with subtitle and pages in parent record', () => {
    const parentRecord = fromJS([
      {
        title: 'A title of book',
        subtitle: 'A subtitle',
        page_start: '1',
        page_end: '10',
        record: { $ref: 'http://localhost:5000/api/literature/1234' },
      },
    ]);
    const wrapper = shallow(<ParentRecordInfo parentRecord={parentRecord} />);
    expect(wrapper).toMatchSnapshot();
  });
});
