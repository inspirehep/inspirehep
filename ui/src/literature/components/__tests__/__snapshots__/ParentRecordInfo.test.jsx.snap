// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentRecordInfo renders with parent record 1`] = `
<div>
  <span>
    Part of 
  </span>
  <Link
    to="/literature/1234"
  >
    A title of book
  </Link>
</div>
`;

exports[`ParentRecordInfo renders with parent record 2`] = `
<div>
  <span>
    Part of 
  </span>
  <Link
    to="/literature/1234"
  >
    A title of book
    <span>
       : 
      A subtitle
    </span>
  </Link>
</div>
`;
